netsim/data/types.py

```diff
@@ -669,6 +683,15 @@ def check_int_value(value: int, xform_int: typing.Optional[typing.Callable]) ->
       else:
         return { '_valid': True }
 
+  def check_reserved_range(
+        ranges: dict,
+        p_addr: typing.Any) -> typing.Optional[str]:
+    for k,v in ranges.items():
```

Comment:
Suggestion to remove 'ranges' as a parameter and simply do:
```
def check_reserved_range( p_addr: typing.Any) -> typing.Optional[str]:
    for k,v in RESERVED_PREFIX[af].items():
```

Reply:
Call me old-fashioned, but I don't like references to variables in a hard-to-figure-out scope (like outer functions) 🤷‍♂️

Reply:
> Call me old-fashioned, but I don't like references to variables in a hard-to-figure-out scope (like outer functions) 🤷‍♂️

I took the use of nested functions as you getting less old-fashioned by the PR ;)

Alternatively, we could define a function near the RESERVED_PREFIXES:
```
def check_reserved_range(af: str, p_addr: typing.Any) -> typing.Optional[str]:
    for k,v in RESERVED_PREFIXES[af].items():
```

(I didn't like the 'global RESERVED_PREFIXES' at the top, only to be used way down)